trin-core/src/portalnet/mod.rs

```diff
@@ -68,8 +113,47 @@ impl ssz::Decode for U256 {
 #[cfg(test)]
 mod test {
     use super::*;
+    use rstest::rstest;
     use ssz::{Decode, Encode};
 
+    const U256_MAX: U256 = U256([u64::MAX, u64::MAX, u64::MAX, u64::MAX]);
+    const U512_MAX: U512 = U512([u64::MAX, u64::MAX, u64::MAX, u64::MAX, 0, 0, 0, 0]);
+
+    #[rstest]
+    #[case(U256([1,2,3,4]), U512([1,2,3,4,0,0,0,0]))]
+    #[case(U256([492,4039,129,9984]), U512([492,4039,129,9984,0,0,0,0]))]
+    #[case(U256_MAX, U512_MAX)]
+    #[case(U256([u64::MAX,u64::MAX,u64::MAX-1,u64::MAX]), U512([u64::MAX,u64::MAX,u64::MAX-1,u64::MAX,0,0,0,0]))]
+    #[case(U256([0,0,0,0]), U512([0,0,0,0,0,0,0,0]))]
+    fn u256_to_u512_test(#[case] input: U256, #[case] expected: U512) {
+        let result = U512::from(input);
+        assert_eq!(expected, result);
+    }
+
+    #[rstest]
+    #[case(U512([1,2,3,4,0,0,0,0]), U256([1,2,3,4]))]
+    #[case(U512([420,69,420,69,0,0,0,0]), U256([420,69,420,69]))]
```

Comment:
Sorry for going full boomer, and tbf I do this kinda thing constantly when I'm developing locally, but I'd rather not push any meme code to master if you can replace this test with something more boring.

Reply:
haha yes i will change that